packages/realm-web/src/environment.ts

```diff
@@ -0,0 +1,80 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { Storage } from "./storage";
+import { Window } from "./OAuth2Helper";
+
+/**
+ * A simplified interface for the console.
+ * This should reflect the lowest common denominator across environemnts.
+ */
+interface Console {
+    /**
+     *
+     */
+    error(...data: any[]): void;
+    /**
+     *
+     */
+    log(...data: any[]): void;
+    /**
+     *
+     */
+    warn(...data: any[]): void;
+}
+
+export type Environment = {
+    /**
+     * A console object used to log stuff.
+     */
+    console: Console;
+    /**
+     * The default storage instance on the environment.
+     */
+    defaultStorage: Storage;
+
+    /**
+     * Open a browser window.
+     */
+    openWindow: (url: string) => Window | null;
+};
+
+let environment: Environment | null = null;
```

Comment:
I'm not sure it is better to export the getter+setter vs just exporting this variable.

Reply:
I want to ensure no code is accessing this if it has not been initialized, so to prevent a strange runtime error I expose it via a function that throws a meaningful error if that invariant is violated.

Reply:
Doesn't seem like a huge improvement over just getting the null error, but 🤷‍♂️